src/main/java/org/dependencytrack/tasks/OSVDownloadTask.java

```diff
@@ -0,0 +1,199 @@
+                VulnerableSoftware vs = mapVulnerabilityToVulnerableSoftware(qm, osvVulnerability);
+                if (vs != null) {
+                    vsList.add(vs);
+                }
+            }
+            LOGGER.debug("Updating vulnerable software for OSV advisory: " + advisory.getId());
+            qm.persist(vsList);
+            synchronizedVulnerability.setVulnerableSoftware(vsList);
+            qm.persist(synchronizedVulnerability);
+        }
+        Event.dispatch(new IndexEvent(IndexEvent.Action.COMMIT, Vulnerability.class));
+    }
+
+    private Vulnerability mapAdvisoryToVulnerability(final QueryManager qm, final OSVAdvisory advisory) {
+
+        final Vulnerability vuln = new Vulnerability();
+        vuln.setSource(Vulnerability.Source.GOOGLE);
+        vuln.setVulnId(String.valueOf(advisory.getId()));
+        vuln.setTitle(advisory.getSummary());
+        vuln.setDescription(advisory.getDetails());
+        vuln.setPublished(Date.from(advisory.getPublished().toInstant()));
+        vuln.setUpdated(Date.from(advisory.getModified().toInstant()));
+
+        if (advisory.getReferences() != null && advisory.getReferences().size() > 0) {
+            final StringBuilder sb = new StringBuilder();
+            for (String ref : advisory.getReferences()) {
+                sb.append("* [").append(ref).append("](").append(ref).append(")\n");
+            }
+            vuln.setReferences(sb.toString());
+        }
+
+        if (advisory.getCweIds() != null) {
+            for (int i=0; i<advisory.getCweIds().size(); i++) {
+                final Cwe cwe = CweResolver.getInstance().resolve(qm, advisory.getCweIds().get(i));
+                if (cwe != null) {
+                    vuln.addCwe(cwe);
+                }
+            }
+        }
+
+        if (advisory.getSeverity() != null) {
+            if (advisory.getSeverity().equalsIgnoreCase("CRITICAL")) {
+                vuln.setSeverity(Severity.CRITICAL);
+            } else if (advisory.getSeverity().equalsIgnoreCase("HIGH")) {
+                vuln.setSeverity(Severity.HIGH);
+            } else if (advisory.getSeverity().equalsIgnoreCase("MODERATE")) {
+                vuln.setSeverity(Severity.MEDIUM);
+            } else if (advisory.getSeverity().equalsIgnoreCase("LOW")) {
+                vuln.setSeverity(Severity.LOW);
+            } else {
+                vuln.setSeverity(Severity.UNASSIGNED);
+            }
+        } else {
+            vuln.setSeverity(Severity.UNASSIGNED);
+        }
+        return vuln;
+    }
+
+    private VulnerableSoftware mapVulnerabilityToVulnerableSoftware(final QueryManager qm, final OSVVulnerability vuln) {
+
+        List<String> versions = vuln.getVersions();
+        String versionStartIncluding = null;
+        String versionEndIncluding = null;
+        if (versions != null) {
+            versionStartIncluding = versions.get(0);
+            versionEndIncluding = versions.get(versions.size() - 1);
```

Comment:
Looking at the example advisory GHSA-vvw4-rfwf-p6hx from OSV that you posted, it seems like we can't rely upon the fact that the latest included version is always the last in the array. 

```json
    "versions": [
        "9.0.0.M1",
        "9.0.0.M10",
        "9.0.0.M11",
...
        "9.0.37",
        "9.0.38",
        "9.0.4",
        "9.0.5",
        "9.0.6",
        "9.0.7",
        "9.0.8"
      ],
```

The possibility of there being arbitrary suffixes (and possibly prefixes) makes sorting a little tricky. 

It may be easier to use the `ranges` object instead. But can we rely on it being always present?

```json
    "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "9.0.0-M1"
            },
            {
              "fixed": "9.0.39"
            }
          ]
        }
    ],
```

`introduced` would map to `versionStartIncluding` and `fixed` to `versionEndExcluding`.

Reply:
I found a case where for a affected package (for a purl), ranges have multiple introduced-fixed pairs like : 
`"ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "2.2"
            },
            {
              "fixed": "2.2.26"
            },
            {
              "introduced": "3.2"
            },
            {
              "fixed": "3.2.11"
            },
            {
              "introduced": "4.0"
            },
            {
              "fixed": "4.0.1"
            }
          ]
        }
      ]`

What should be versionStartIncluding and versionEndExcluding in this case?
@nscuro @VinodAnandan

Reply:
OSV's API methods need whole list of packages being used as input and returns same json with same ranges format.
Yes, we can create multiple VulnerableSoftware instances, one for each range for a vulnerability object. This seems like a reliable option for now.

Reply:
@sahibamittal First off. I want to thank you for such a quality PR. Very much appreciated.

Were you able to create multiple VulnerableSoftware instances yet? One for each range?